api/announcements.py

```diff
@@ -13,9 +15,20 @@ class Announcements:
     """
 
     SETTING_NAME = "announcements"
+    # Must keep the global and library local settings names different
+    # to avoid configuration inheritance
+    GLOBAL_SETTING_NAME = "global_announcements"
 
     @classmethod
-    def for_library(cls, library):
+    def for_all(cls, _db) -> "Announcements":
```

Comment:
You should be able to use the `Announcements` class name (rather than a string) here by importing the `annotations` future at the top of the file.